torch/_dynamo/source.py

```diff
@@ -178,6 +178,18 @@ def name(self):
         else:
             raise AssertionError(f"unhandled {self.prop}")
 
+    def __eq__(self, other):
+        return (
+            self.base == other.base
+            and self.prop == other.prop
+            and self.idx == other.idx
+        )
+
+    # TODO(voz): This isnt really used in any way where we care about having a real hash yet.
+    # This is just to be able to chuck these things in a set.
+    def __hash__(self):
+        return id(self)
```

Comment:
This seems bad. If we really want hash, I'd suggest we start marking these Sources as frozen=True so they get hashes automatically generated. Best to do separately.

Reply:
We could also use a list, there is not a ton of these :)